feat: t10 - implement your test host

rtest12 opened the pull request:
## [Link to the task](https://github.com/saritasa-nest/saritasa-devops-camp/blob/main/chapter3%20-%20tech-stack%20components/l1/l1%20-%20nginx.md#implement-your-test-host)

## implement your test host

- setup nginx supporting 2 servers
- camp-php.local (this will be used for future php experiments)
- camp-python.local (this will be used for future python experiments)
- both servers should support named certificates for the domains mentioned above. use mkcert cli to do that
- both server should implement custom location /hello, that will return 200 OK status code along with the value of the query_string passed into that URL, [see here](https://www.techopedia.com/definition/1228/query-string), for example https://camp-php.local/hello?tech_stack=php, which should output the entire query_string
- your SSL cert should be valid in CURL call in CLI, as well as if you open it in the browser
- your camp-php.local server should output logs in JSON format
- open PR when you done, in the PR - pls demonstrate the output of the curl call for the URL in each domain, and the screenshot in the browser

Curl log `https://camp-php.local/`

[curl_camp_php.log](https://github.com/rtest12/devops-camp-2023/files/11219386/curl_camp_php.log)

Curl log `https://camp-python.local/`

[curl_camp_python.log](https://github.com/rtest12/devops-camp-2023/files/11219387/curl_camp_python.log)

Nginx json log:

[access.json.log](https://github.com/rtest12/devops-camp-2023/files/11219401/access.json.log)

Screenshots:
![Screenshot from 2023-04-12 09-12-12](https://user-images.githubusercontent.com/93929010/231367177-9c4d8ec2-6c04-4545-a29c-2cfbbbe3dddc.png)

![Screenshot from 2023-04-12 09-12-37](https://user-images.githubusercontent.com/93929010/231367209-ac82455b-629d-4460-bae8-a1a0c8499130.png)

![Screenshot from 2023-04-12 09-18-45](https://user-images.githubusercontent.com/93929010/231368272-85f5b0e7-ce3b-4ae7-95d8-7cf6ac4f5b42.png)

![Screenshot from 2023-04-12 09-19-03](https://user-images.githubusercontent.com/93929010/231368290-30741402-23f3-4dfa-a74b-936949a29a37.png)

Commit:
add curl.log

dmitry-mightydevops reviewed (changes requested):
I only see curl.log in your PR :)

dmitry-mightydevops commented on chapter2/tests/mkcert/curl.log:
such file(s) is a noise - don't store it in the remote, if you need - you can always attach files to PR when you open it.

rtest12 replied:
> such file(s) is a noise - don't store it in the remote, if you need - you can always attach files to PR when you open it.

Ok, should all screenshots and duplicate logs be removed from the description now?

dmitry-mightydevops replied:
@rtest12 
summary is OK to have any artifacts, but the actual repo should not have any log artifacts committed.

dmitry-mightydevops commented:
@rtest12 also change the title of the PR to be consistent with other PRs.

dmitry-mightydevops commented:
@rtest12 also use proper `feature/branch` names for your branches. `mkcert` is not a good example and is not semantic.

`feature/add-php-python-sample-nginx-configs`
or
`feature/add-sample-nginx-configs`

is much better name.

Naming is important.

rtest12 commented:
> @rtest12 also change the title of the PR to be consistent with other PRs.

Fixed, I thought about it, I was confused by the title of the task. I should have clarified those details right away.

Commit:
add logs and screenshots

dmitry-mightydevops commented:
> > @rtest12 also change the title of the PR to be consistent with other PRs.
> 
> Fixed, I thought about it, I was confused by the title of the task. I should have clarified those details right away.

@rtest12 
you titlte should be always lowercase

dmitry-mightydevops reviewed (changes requested):
your task has invalid artifacts. I don't need png files or logs inside the repo.
repo should have only the solution, which you didn't commit.

Where is nginx.conf (and/or all other associated files)?

Commit:
add conf nginx files

rtest12 commented:
> your task has invalid artifacts. I don't need png files or logs inside the repo. repo should have only the solution, which you didn't commit.
> 
> Where is nginx.conf (and/or all other associated files)?

@dmitry-mightydevops 

I probably took the task too literally, since it didn’t indicate that these files were needed, I thought it was strange, but it means you just need a result)) I fixed everything, uploaded the files, deleted unnecessary ones, sorry))

dmitry-mightydevops commented on chapter2/tests/mkcert/nginx.conf:
```diff
@@ -0,0 +1,87 @@
+
+http {
+
+	##
+	# Basic Settings
+	##
```
such comment style make reading much harder. Your comments should provide some values, but right now they're too basic to provide any value.

dmitry-mightydevops commented on chapter2/tests/mkcert/nginx.conf:
```diff
@@ -0,0 +1,87 @@
+
+events {
+	worker_connections 768;
+	# multi_accept on;
```
If you don't need specific settings - don't keep them or commented, this would lead to a chaos later on during the maintenance phase.

Why he/she did comment it? Was it useful for something? For what?
People ask too many questions in such cases.

dmitry-mightydevops commented on chapter2/tests/mkcert/nginx.conf:
```diff
@@ -0,0 +1,87 @@
+	# Logging Settings
+	##
+
+	log_format logger-json escape=json '{"source": "nginx", "time": $msec, "resp_body_size": $body_bytes_sent, "host": "$http_host", "address": "$remote_addr", "request_length": $request_length, "method": "$request_method", "uri": "$request_uri", "status": $status,  "user_agent": "$http_user_agent", "resp_time": $request_time, "upstream_addr": "$upstream_addr"}';
```
format this better. Hard to read. Use multiline approach.

dmitry-mightydevops commented on chapter2/tests/mkcert/nginx.conf:
```diff
@@ -0,0 +1,87 @@
+
+	gzip on;
+
+	# gzip_vary on;
+	# gzip_proxied any;
+	# gzip_comp_level 6;
+	# gzip_buffers 16 8k;
+	# gzip_http_version 1.1;
+	# gzip_types text/plain text/css application/json application/javascript text/xml application/xml application/xml+rss text/javascript;
+
+	##
+	# Virtual Host Configs
+	##
```
same issues

dmitry-mightydevops commented on chapter2/tests/mkcert/nginx.conf:
```diff
@@ -0,0 +1,87 @@
+}
+
+
+#mail {
+#	# See sample authentication script at:
+#	# http://wiki.nginx.org/ImapAuthenticateWithApachePhpScript
+#
+#	# auth_http localhost/auth.php;
+#	# pop3_capabilities "TOP" "USER";
+#	# imap_capabilities "IMAP4rev1" "UIDPLUS";
+#
+#	server {
+#		listen     localhost:110;
+#		protocol   pop3;
+#		proxy      on;
+#	}
+#
+#	server {
+#		listen     localhost:143;
+#		protocol   imap;
+#		proxy      on;
+#	}
+#}
```
same

dmitry-mightydevops commented on chapter2/tests/mkcert/nginx.conf:
```diff
@@ -0,0 +1,87 @@
+	# Virtual Host Configs
+	##
+
+	include /etc/nginx/conf.d/*.conf;
+	include /etc/nginx/sites-enabled/*;
```
please have a single nginx.conf file, so it's easy to run locally.

dmitry-mightydevops commented on chapter2/tests/mkcert/sites-available/camp-php.local.conf:
```diff
@@ -0,0 +1,31 @@
+		return 200 $query_string;
+        }
+
+}
```
you formatting is OFF. Please use proper indentation and use tabs everywhere

dmitry-mightydevops commented on chapter2/tests/mkcert/sites-available/camp-php.local.conf:
```diff
@@ -0,0 +1,31 @@
+server {
+        if ($host = camp-php.local) {
+            return 301 https://$host$request_uri;
+        }
+
+	    server_name camp-php.local;
+	    listen 80;
+    	root /opt/camp/camp-php/;
+}  
```
no need for port 80 sites.

dmitry-mightydevops commented on chapter2/tests/mkcert/sites-available/camp-php.local.conf:
```diff
@@ -0,0 +1,31 @@
+    	access_log /var/log/nginx/access.log logger-json;
+
+
+        location / {
+	    root  /opt/camp/camp-php/;
+	    index  index.html;
+        }
+
+
+        location /hello {
+		default_type text/html;
+		return 200 $query_string;
+        }
```
indentation you have is like this
![image](https://user-images.githubusercontent.com/70822710/231887776-bf7cea9c-5228-4dd3-9b36-c44e59d1b398.png)

dmitry-mightydevops commented on chapter2/tests/mkcert/sites-available/camp-python.local.conf:
```diff
@@ -0,0 +1,28 @@
+server {
+        if ($host = camp-python.local) {
+            return 301 https://$host$request_uri;
+        }
+
+        server_name camp-python.local;
+        listen 80;
+        root /opt/camp/camp-php/;
+        }
+server {
+        server_name camp-python.local;
+        listen 443 ssl;
+        ssl_certificate "/opt/certs/camp-python.local+4.pem";
+        ssl_certificate_key "/opt/certs/camp-python.local+4-key.pem";
+
+
+        location / {
+                root  /opt/camp/camp-python/;
+                index  index.html;
+        }
+
+
+location /hello {
+                default_type text/html;
+                return 200 $query_string;
+                }
+
+	}
```
same issues as above.

rtest12 replied:
@dmitry-mightydevops 
Fixed

Commit:
style fix and delete includes

Commit:
style fix and delete includes

Commit:
push error

Commit:
Delete unnecessary files

Commit:
style fix
